Create polls.md (#47)
ScootKit/scnx-docs: i18n/de/docusaurus-plugin-content-docs/current/custom-bot/modules/community/polls.md

# Umfragen

Einfaches Modul zum Erstellen neuer Umfragen auf deinem Server!

<ModuleOverview moduleName="polls" />

## Funktionen {#features}

* Erstelle anonyme und öffentliche Umfragen mit bis zu neun Optionen pro Umfrage.
* Nutzer können via ein Dropdown-Menü ihre Option auswählen.

> * Automatisch nach Ablauf einer festgelegten Zeit endende Umfragen.

Author: done

* Leicht verständliche Ergebnisanzeige, welche live aktualisiert wird.

Hier ist ein Beispiel einer beendeten Umfrage:

![](@site/docs/assets/custom-bot/modules/polls/example.png)

## Einrichtung {#setup}

* Um mit dem Modul Umfragen zu starten, stelle sicher, dass dein Bot in jedem Kanal, in dem du Umfragen verwenden willst,
  folgende Rechte hat: Nachrichten lesen, Nachrichten senden, Links einbetten.
* Stelle sicher, dass du [die Rechte](./../../slash-commands) für [`/poll`](#commands) korrekt eingestellt hast.

## Verwendung {#usage}

Administratoren können Umfragen mit folgenden Befehlen verwalten:

> * [`/poll create`](#commands) zum erstellen neuer Umfragen.

Author: done

| Embed | Mit dieser Konfigurationsoption kannst Du anpassen, wie das Umfrage-Embed in Discord angezeigt wird. Die folgenden Werte können angepasst werden: <ul><li><code>title</code>: Der Name des Embeds.</li><li><code>color</code>: Die Farbe des Embeds, muss ein <a href="../../additional-features#embed-colors">unterstützter Farbwert</a> sein.</li><li><code>options</code>: Die Überschrift des Optionsfelds im Embed.</li><li><code>liveView</code>: Die Überschrift der (Live-)Ergebnisanzeige im Embed.</li><li><code>expiresOn</code>: Die Überschrift des Embeds, wenn die Umfrage beendet wurde.</li><li><code>endedPollTitle</code>: Die Überschrift des Embeds, wenn die Umfrage beendet wurde.</li><li><code>endedPollColor</code>: Die Farbe des Embeds, wenn die Umfrage beendet ist. Muss ein <a href="../../additional-features#embed-colors">unterstützter Farbwert</a> sein.</li><li><code>visibility</code>: Die Überschrift des Öffentlichkeitsabschnitts des Embeds.</li></ul> |

## Fehlerbehebung {#troubleshooting}

Wenn du Probleme mit dem Umfragemodul hast, stelle bitte sicher, dass

> * Du hast keine Umfrage in einem Ankündigungskanal gestartet hast (da Nachrichten in solchen Kanälen nicht so häufig aktualisiert werden können).

Author: done

> * deine [Nachrichtenkonfiguration](#configuration-strings) keine leeren Felder enthält, und das [Farbwerte gültig sind](../../additional-features#embed-colors).    

Author: done

> * Ihre [Nachrichtenkonfiguration](#configuration-strings) keine leeren Felder enthält, und das [Farbwerte gültig sind](../../additional-features#embed-colors).    

Author: done

> * deine [Konfiguration](#configuration-config) nur gültige Emoji-Werte (Unicode für integrierte Emojis und Emoji-IDs für benutzerdefinierte Emojis)enthält .

Reviewer: ```suggestion
* deine [Konfiguration](#configuration-config) nur gültige Emoji-Werte (Unicode für integrierte Emojis und Emoji-IDs für benutzerdefinierte Emojis) enthält .
```

Author: done

> * deine [Konfiguration](#configuration-config) nur gültige Emoji-Werte (Unicode für integrierte Emojis und Emoji-IDs für benutzerdefinierte Emojis) enthält .

Reviewer 2: ```suggestion
* deine [Konfiguration](#configuration-config) nur gültige Emoji-Werte (Unicode für integrierte Emojis und Emoji-IDs für benutzerdefinierte Emojis) enthält.
```

Author: done

> * Ihre [Konfiguration](#configuration-config) nur gültige Emoji-Werte (Unicode für integrierte Emojis und Emoji-IDs für benutzerdefinierte Emojis)enthält .

Author: done

> * Der Bot über die folgenden Berechtigungen für den Kanal, in dem Du eine Umfrage starten möchten: Nachrichten lesen, Nachrichten senden, Links einbetten verfügt.

Reviewer: ```suggestion
* der Bot über die folgenden Berechtigungen in dem Kanal, in dem du eine Umfrage starten möchtest, verfügt: Nachrichten lesen, Nachrichten senden, Links einbetten.
```

Author: done

## Gespeicherte Daten {#data-usage}

> Die folgenden Daten werden über jede Umfrage gespeichert, die über [Befehle](#commands) erstellt wird :

Reviewer 2: ```suggestion
Die folgenden Daten werden über jede Umfrage gespeichert, die über [Befehle](#commands) erstellt wird:
```

Author: done

> * Die ID der Umfragenachricht, die nach der Erstellung gesendet wurde
> * Die ID des Kanals, in dem die Umfrage erstellt wurde
> * Die Frage der Umfrage (`description`-Wert im Befehl), inkludiert, ob die Umfrage öffentlich ist oder nicht
> * Der Wert jeder Abstimmungsoption (`option{n}`-Wert im Befehl)
> * Falls aktiviert: Genaue Uhrzeit, zu der die Umfrage abläuft (gesteuert durch den `duration`-Wert im Befehl) und ob die Umfrage beendet ist
> * Metadaten über Ihren Server anzeigen (z. B. Name, Symbol, Besitzer, ...)

Reviewer 2: ```suggestion
* Die eindeutige Identifikationsnummer der Umfragenachricht, die nach der Erstellung gesendet wurde
* Die eindeutige Identifikationsnummer des Kanals, in dem die Umfrage erstellt wurde
* Die Frage der Umfrage (`description`-Wert im Befehl), inkludiert, ob die Umfrage öffentlich ist oder nicht
* Der Wert jeder Abstimmungsoption (`option{n}`-Wert im Befehl)
* Falls aktiviert: Genaue Uhrzeit, zu der die Umfrage abläuft (gesteuert durch den `duration`-Wert im Befehl) und ob die Umfrage beendet ist
* Metadaten über deinen Server anzeigen (z. B. Name, Symbol, Besitzer, ...)
```

Author: done

Darüber hinaus werden die folgenden Daten über jeden Benutzer gespeichert, wenn er für eine Option abstimmt:

> * Die ID des Benutzers

Reviewer 2: ```suggestion
* Die eindeutige Identifikationsnummer des Benutzers
```

Author: done

# Umfragen

> Einfaches Modul zum Erstellen neuer Umfragen auf Ihrem Server!

Author: Ok thanks 

you write me that in dc

> Einfaches Modul zum Erstellen neuer Umfragen auf Ihrem Server!

Author: done

<ModuleOverview moduleName="polls" />

> ## Funktionen {#funktionen}

Author: ok i'will do it

* Erstelle anonyme und öffentliche Umfragen mit bis zu neun Optionen pro Umfrage.
* Nutzer können via ein Dropdown-Menü ihre Option auswählen.

> * Automatisch endende Umfragen, nach Ablauf einer festgelegten Zeit.

Reviewer: ```suggestion
* Automatisch nach Ablauf einer festgelegten Zeit endende Umfragen.
```

Author: done

> * Leicht verständliche Ergebnisanzeige, welche live geupdatet wird.

Reviewer: ```suggestion
* Leicht verständliche Ergebnisanzeige, welche live aktualisiert wird.
```

Author: done

> Hier ist ein Beispiel, einer beendeten Umfrage:

Reviewer: ```suggestion
Hier ist ein Beispiel einer beendeten Umfrage:
```

Author: done

> Hier ist ein Beispiel, wenn eine Umfrage beendet, ist:

Author: ok

> Hier ist ein Beispiel, wenn eine Umfrage beendet, ist:

Author: done

![](@site/docs/assets/custom-bot/modules/polls/example.png)

> ## Einrichtung {#einrichtung}

Author: ok

> * Um mit dem Modul Umfragen zu starten, stelle sicher, dass dein Bot in jedem Channel, in dem du Umfragen verwenden willst,

Reviewer: ```suggestion
* Um mit dem Modul Umfragen zu starten, stelle sicher, dass dein Bot in jedem Kanal, in dem du Umfragen verwenden willst,
```

Author: done

  folgende Rechte hat: Nachrichten lesen, Nachrichten senden, Links einbetten.

> * Stelle sicher, dass Du [die Rechte](./../../slash-commands) für [`/poll`](#commands) korrekt eingestellt haben.

Reviewer: ```suggestion
* Stelle sicher, dass du [die Rechte](./../../slash-commands) für [`/poll`](#commands) korrekt eingestellt hast.
```

## Verwendung {#usage}

> Administratoren können Umfragen mit folgenden Commands managen:

Reviewer: ```suggestion
Administratoren können Umfragen mit folgenden Befehlen verwalten:
```

Author: done

> * [`/poll create`](#commands) Zum Erstellen neuer Umfragen.

Reviewer: ```suggestion
* [`/poll create`](#commands) zum Erstellen einer neuen Umfrage.
```

> * [`/poll end`](#commands) Zum Beenden einer Umfrage (Umfragen enden automatisch, wenn du `/poll create` den Dauerparameter gesetzt haben).

Reviewer: ```suggestion
* [`/poll end`](#commands) zum Beenden einer Umfrage (Umfragen enden automatisch, wenn du bei `/poll create` den Zeitparameter gesetzt hast).
```

Author: done

Benutzer können:

> * Über Umfragen abstimmen, unter Verwendung des Auswahl Menüs unter der Umfrage.

Reviewer: ```suggestion
* Mit dem Dropdown-Menü unter Umfragen abstimmen.
```

Author: done

> * Ihre Antwort einsehen, für die Du gestimmt haben, über den Knopf „Für was habe ich abgestimmt“ unter der Abstimmung.

Reviewer: ```suggestion
* Sehen, für was sie gestimmt haben, indem sie den Knopf „Für was habe ich abgestimmt?“ unter der Abstimmung nutzen.
```

Author: done

> * Ihre Stimme entfernen, wenn Du zuerst „Für was habe ich gestimmt“ ausgewählt haben und danach auf „Meine Stimme entfernen“ unter der Rückmeldung klicken.

Reviewer: ```suggestion
* Ihre Stimme entfernen, wenn sie zuerst „Für was habe ich abgestimmt?“ auswählen und danach auf „Meine Stimme entfernen“ unter der Antwort klicken.
```

Author: done

> * Liveergebnisse im Umfragen Embed sehen.

Reviewer: ```suggestion
* Liveergebnisse im Umfragen-Embed sehen.
```

Author: done

> * Sich anschauen, wer für welche Option gestimmt hat, indem Du auf den Button „Aktuelle
>   Stimmen anzeigen“ unter dem Abstimmungs-Embed klicken (falls die Umfrage auf Öffentlich gestellt ist).

Reviewer: ```suggestion
* Sich anschauen, wer für welche Option gestimmt hat, indem sie auf den Knopf „Aktuelle
  Stimmen anzeigen“ unter dem Abstimmungs-Embed klicken (falls die Umfrage auf öffentlich gestellt ist).
```

Author: done

|--------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `/polls create description:<Text> channel:<TextChannel> option1:<Text> option2:<Text> [option{n}:<Text>] [duration:<TimeDuration>] [public:<Boolean>]` | Erstelle im gewünschten Kanal eine neue Umfrage. Gib die Umfragefrage als `description` Wert ein und füge mögliche Optionen mit dem Element `option{n}` hinzu. Optional kannst Du eine Umfrage mit der Option `public` öffentlich machen (das bedeutet, dass jeder sehen kann, wofür Benutzer gestimmt haben). Wenn diese Option aktiviert ist, wird unterhalb der Umfrage die Schaltfläche „Aktuelle Stimmen anzeigen“ hinzugefügt. Um eine Umfrage automatisch zu beenden, gibst Du eine `duration` in einem [unterstützten Zeitdauerwert](./../../additional-features#durations) an. |

Reviewer: ```suggestion
| `/polls create description:<Text> channel:<TextKanal> option1:<Text> option2:<Text> [option{n}:<Text>] [duration:<ZeitDauer>] [public:<Boolean>]` | Erstelle im gewünschten Kanal eine neue Umfrage. Gib die Umfragefrage als `description` Wert ein und füge mögliche Optionen mit dem Element `option{n}` hinzu. Optional kannst Du eine Umfrage mit der Option `public` öffentlich machen (das bedeutet, dass jeder sehen kann, wofür Benutzer gestimmt haben). Wenn diese Option aktiviert ist, wird unterhalb der Umfrage die Schaltfläche „Aktuelle Stimmen anzeigen“ hinzugefügt. Um eine Umfrage automatisch zu beenden, gibst du eine `duration` in einem [unterstützten Zeitdauerwert](./../../additional-features#durations) an. |
```

Author: done

> | `/polls end msg-id:<PollMessageID>`                                                                                                                    | Beendet eine bereits bestehende Umfrage, die noch nicht beendet wurde, sofort. Die Option `msg-id` unterstützt die automatische Vervollständigung, sodass Du problemlos nach der Umfragebeschreibung der Umfrage suchen können, die Du beenden möchten.                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| `/polls end msg-id:<UmfragenNachrichtID>`                                                                                                                    | Beendet sofort eine bereits bestehende Umfrage, die noch nicht beendet wurde. Die Option `msg-id` unterstützt die automatische Vervollständigung, sodass du problemlos nach der Umfragebeschreibung der Umfrage suchen kannst, die du beenden möchtest.                                                                                                                                                                                                                        |
```

Author: done

## Konfiguration {#configuration}

### Konfiguration {#configuration-config}

> In dieser Datei kannst du die Emojis neben den Abstimmungsoptionen anpassen. Öffne es in 
> deinem [Dashboard](https://scnx.app/de/glink?page=bot/configuration?query=poll&file=polls|configs/config).

Reviewer: ```suggestion
In dieser Datei kannst du die Emojis neben den Abstimmungsoptionen anpassen. Öffne sie in 
deinem [Dashboard](https://scnx.app/de/glink?page=bot/configuration?query=poll&file=polls|configs/config).
```

Author: done

> In dieser Datei kannst DU die Emojis neben den Abstimmungsoptionen anpassen. Öffne es in 
> Ihrem [Dashboard](https://scnx.app/de/glink?page=bot/configuration?query=poll&file=polls|configs/config).

Reviewer: ```suggestion
In dieser Datei kannst du die Emojis neben den Abstimmungsoptionen anpassen. Öffne sie in 
deinem [Dashboard](https://scnx.app/de/glink?page=bot/configuration?query=poll&file=polls|configs/config).
```

Author: done

| Feld  | Beschreibung                                                                                                                                                                                                                                            |
|--------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Emojis | In diesen Feldern kannst du die Emojis anpassen, die neben jeder Abstimmungsoption im Dropdown-Menü und im Embed angezeigt werden. Stelle sicher, dass Du Unicode-Werte für integrierte Emojis oder gültige Emoji-IDs (nicht nur den Namen) für benutzerdefinierte Emojis eingibst. |

Reviewer: ```suggestion
| Emojis | In diesen Feldern kannst du die Emojis anpassen, die neben jeder Abstimmungsoption im Dropdown-Menü und im Embed angezeigt werden. Stelle sicher, dass du Unicode-Werte für integrierte Emojis oder gültige Emoji-IDs (nicht nur den Namen) für benutzerdefinierte Emojis eingibst. |
```

Author: done

> | Emojis | In diesen Feldern kannst Du die Emojis anpassen, die neben jeder Abstimmungsoption im Dropdown-Menü und in der Einbettung angezeigt werden. Stelle sicher, dass Du Unicode-Werte für integrierte Emojis oder gültige Emoji-IDs (nicht nur den Namen) für benutzerdefinierte Emojis eingeben. |

Reviewer: ```suggestion
| Emojis | In diesen Feldern kannst du die Emojis anpassen, die neben jeder Abstimmungsoption im Dropdown-Menü und im Embed angezeigt werden. Stelle sicher, dass du Unicode-Werte für integrierte Emojis oder gültige Emoji-IDs (nicht nur den Namen) für benutzerdefinierte Emojis eingibst. |
```

Author: done

> ### Nachricht {#configuration-strings}

Reviewer: ```suggestion
### Nachrichten {#configuration-strings}
```

Author: done

> In dieser Datei kannst du anpassen, wie das Umfrage-Embed in Discord angezeigt wird. Öffnen sie es in deinem

Reviewer: ```suggestion
In dieser Datei kannst du anpassen, wie das Umfrage-Embed in Discord angezeigt wird. Öffne sie in deinem
```

Author: done

> In dieser Datei kannst du anpassen, wie die Umfrageeinbettung in Discord angezeigt wird. Öffne es in Deinem
> [Dashboard](https://scnx.app/de/glink?page=bot/configuration?query=poll&file=polls|configs/strings).

Reviewer: ```suggestion
In dieser Datei kannst du anpassen, wie das Umfrage-Embed in Discord angezeigt wird. Öffne sie in deinem
[Dashboard](https://scnx.app/de/glink?page=bot/configuration?query=poll&file=polls|configs/strings).
```

Author: done

|-------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Embed | Mit dieser Konfigurationsoption kannst Du anpassen, wie die Umfrageeinbettung in Discord angezeigt wird. Die folgenden Werte können angepasst werden: <ul><li><code>title</code>: Der Name des Embeds.</li><li><code>color</code>: Die Farbe des Embeds, muss ein <a href="../../additional-features#embed-colors">unterstützter Farbwert</a>.</li><li><code>options</code>: Die Kopfzeile des Optionsfelds im Embed.</li><li><code>liveView</code>: Der Header der (Live-)Ergebnisanzeige in der Einbettung.</li><li><code>expiresOn</code>: Der Header der Anzeige der Ablaufzeit im Embed.</li><li><code>endedPollTitle</code>: Der Header von Embed, wenn die Umfrage beendet wurde.</li><li><code>endedPollColor</code>: Die Farbe des Embeds wenn die Umfrage beendet ist, muss ein <a href="../../additional-features#embed-colors">unterstützter Farbwert</a> sein.</li><li><code>visibility</code>: Die Kopfzeile des Sichtbarkeitsabschnitts der Umfrageeinbettung.</li></ul> |

Reviewer: ```suggestion
| Embed | Mit dieser Konfigurationsoption kannst Du anpassen, wie das Umfrage-Embed in Discord angezeigt wird. Die folgenden Werte können angepasst werden: <ul><li><code>title</code>: Der Name des Embeds.</li><li><code>color</code>: Die Farbe des Embeds, muss ein <a href="../../additional-features#embed-colors">unterstützter Farbwert</a> sein.</li><li><code>options</code>: Die Überschrift des Optionsfelds im Embed.</li><li><code>liveView</code>: Die Überschrift der (Live-)Ergebnisanzeige im Embed.</li><li><code>expiresOn</code>: Die Überschrift der Anzeige der Ablaufzeit im Embed.</li><li><code>endedPollTitle</code>: Die Überschrift des Embeds, wenn die Umfrage beendet wurde.</li><li><code>endedPollColor</code>: Die Farbe des Embeds, wenn die Umfrage beendet ist. Muss ein <a href="../../additional-features#embed-colors">unterstützter Farbwert</a> sein.</li><li><code>visibility</code>: Die Überschrift des Öffentlichkeitsabschnitts des Embeds.</li></ul> |
```

Author: done

## Fehlerbehebung {#troubleshooting}

> Wenn Du Probleme mit dem Umfragemodul haben, stelle bitte sicher, dass

Reviewer: ```suggestion
Wenn du Probleme mit dem Umfragemodul hast, stelle bitte sicher, dass
```

Author: done

* Du hast keine Umfrage in einem Ankündigungskanal gestartet (da Nachrichten in solchen Kanälen nicht so häufig aktualisiert werden können).
* Ihre [Nachrichtenkonfiguration](#configuration-strings) keine leeren Felder enthält, und das [Farbwerte gültig sind](../../additional-features#embed-colors).    
* Ihre [Konfiguration](#configuration-config) onnur gültige Emoji-Werte (Unicode für integrierte Emojis und Emoji-IDs für benutzerdefinierte Emojis)enthält .
* Der Bot über die folgenden Berechtigungen für den Kanal, in dem Du eine Umfrage starten möchten: Nachrichten lesen, Nachrichten senden, Links einbetten verfügt.

## Gespeicherte Daten {#data-usage}

> Die folgenden Daten werden über jede Abfrage gespeichert, die über [Befehle](#commands) erstellt wird :

Reviewer: ```suggestion
Die folgenden Daten werden über jede Umfrage gespeichert, die über [Befehle](#commands) erstellt wird:
```

Author: done

* Die ID der Umfragenachricht, die nach der Erstellung gesendet wurde
* Die ID des Kanals, in dem die Umfrage erstellt wurde

> * Die Frage nach der Umfrage (Wert im Befehl), einschließlich der Frage, ob die Umfrage öffentlich ist oder nicht (`description` value in the command), einschließlich der Frage, ob die Umfrage öffentlich ist oder nicht

Reviewer: ```suggestion
* Die Frage der Umfrage (`description`-Wert im Befehl), inkludiert, ob die Umfrage öffentlich ist oder nicht
```

Author: done

> * Der Wert jeder Abstimmungsoption ( Werte im Befehl) (`option{n}` values in the command)

Reviewer: ```suggestion
* Der Wert jeder Abstimmungsoption (`option{n}`-Wert im Befehl)
```

Author: done

> * Falls aktiviert: Genaue Uhrzeit, zu der die Abfrage abläuft (gesteuert durch die Option im Befehl) und ob Die Umfrage ist beendet (controlled by the `duration` option in the command) und ob
>   die Umfrage ist beendet

Reviewer: ```suggestion
* Falls aktiviert: Genaue Uhrzeit, zu der die Umfrage abläuft (gesteuert durch den `duration`-Wert im Befehl) und ob die Umfrage beendet ist
```

Author: done

* Metadaten über Ihren Server anzeigen (z. B. Name, Symbol, Besitzer, ...)

Darüber hinaus werden die folgenden Daten über jeden Benutzer gespeichert, wenn er für eine Option abstimmt:

* Die ID des Benutzers

> * Die Nummer der Option, für die Du gestimmt haben

Reviewer: ```suggestion
* Die Nummer der Option, für die gestimmt wurde
```

Author: done

> Wenn die Umfrage noch nicht beendet ist, können Benutzer [Ihre Stimme entfernen](#usage). Nach Beendigung einer Umfrage ist diese Option nicht mehr verfügbar.

Reviewer: ```suggestion
Wenn die Umfrage noch nicht beendet ist, können Benutzer [ihre Stimme entfernen](#usage). Nach Beendigung einer Umfrage ist diese Option nicht mehr verfügbar.
```

Author: done

> Es gibt keine Möglichkeit, Umfragen aus der Datenbank zu entfernen (Du kannst die Umfrage trotzdem jederzeit löschen). Zum
> löschen aller Umfragen, [lösche die Moduldatenbank](./../../additional-features#reset-module-database).

Reviewer: ```suggestion
Es gibt keine Möglichkeit, Umfragen aus der Datenbank zu entfernen (du kannst die Umfrage trotzdem jederzeit löschen). Zum
Löschen aller Umfragen, [lösche die Moduldatenbank](./../../additional-features#reset-module-database).
```

Author: done
